woocommerce/woocommerce-gateway-stripe
# Update shipping methods request

Fixes #755, fixes #1177 .

When `requestShipping` option is enabled on payment request the following happens:

1. When on a cart page and shipping options selected (address and method).
2. User clicks Apple Pay / Google pay button (doesn't matter to be honest).
3. Popup is opened and default shipping address (might be the same or different) configured in Apple Pay is applied. Hence new shipping options for a selected address should are fetched.
4. Previously selected shipping cost displayed while new shipping methods are being fetched.
5. Once fetched, first in the list shipping option is selected by default.

I.e. step 3 triggers a shipping address change flow in the same way as normal cart UI does.

#### Changes proposed in this Pull Request:
- Keep the previously selected shipping method by sorting shipping options.

#### Testing instructions

Preparation:

* Add some real card to Apple pay on Mac and Google Pay in Chrome.
* Use ngrok and Safari for Apple Pay testing to access the site with https.
* Create at least two shipping methods for shipping zone.

Testing (same for Apple Pay and Google Pay):

1. Add product to the cart.
2. Go to the cart page.
3. Select some shipping options (address/method)
4. Click Apple Pay button. All shipping details should be correct.
5. Confirm payment. The order should be placed successfully.

-------------------
